### ISSUE TITLE
[0.10] deep associations

### ISSUE THREAD

**Issue by Haks1** [balderdashy/sails-mongo#108]
Hi guys !

Ideally, I wanted to use this mongo schema (simplyfied here):

```
Collection Project
{
    name: string,
 /*
 ...
 other vals
 ...
 */
   emptimes:
         {
           /*
           ...
           other vals
           ...
           */ 
        emps: model collection Employee (MtoM),
        timespans
        {
                start: date,
                end: date
        }
    }
}

---------------------------------------------
---------------------------------------------

Collection Employee
{
    name: string,
 /*
 ...
 other vals
 ...
 */
        projs: model collection Project (MtoM),
}
```

I understood that embedded documents are not supported actually, as also associations with specific data binded to it, so I made my mongo "schema" as relational DB:

```
Project attributes: {
  	
    name:
        {
        type: 'string',
        required: true,
        alpha: true,
        maxLength: 30,
        minLength: 3
        },

 /*
 ...
 other vals
 ...
 */

    collabs: 
        {
            collection: 'ProjectEmployee',
            via: 'proj'
        }
  }

---------------------------------------------
---------------------------------------------

Employee.attributes: {
  	
    nickname: {
        type: 'string',
        required: true,
        unique: true,
        alpha: true,
        maxLength: 30,
        minLength: 3
        },

 /*
 ...
 other vals
 ...
 */

    projs: 
    {
        collection: 'ProjectEmployee',
        via: 'collab'
    }
    

  }

---------------------------------------------
---------------------------------------------

ProjectEmployee.attributes: {

			collab: 
            {
                model:'Employee'
            },

			proj: 
            {
				model:'Project'
            },

  /*
 ...
 other vals
 ...
 */

            spans:
	        {
	            collection: 'CalSpan',
            	via: 'projCollab'
	        }
	}


---------------------------------------------
---------------------------------------------

CalSpan.attributes: {

				projCollab: 
	            {
					model:'ProjetEmployee'
	            },

  /*
 ...
 other vals
 ...
 */
	}
```

My question is now simple: how can I populate the deep array branch of my json tree went I start with just the collection Project ( Project.find({}) ) ?
For example, How do I get the employees collection from a specific project ? Is it possible in one shot ?

**Comment by arnthorsnaer** [comment]
+1

**Comment by riblee** [comment]
+1

**Comment by laurentjacob** [comment]
+1

**Comment by dmarcelino** [comment]
> We are working on a populate command that takes the dot notation:
> 
> ```javascript
> .populate('pet')
> .populate('pet.owner')
> ```
> 
> It's a little more complicated for sequel databases but in-memory operations for databases that don't support joins (like mongo) should be a little easier.

That feature is being tracked in balderdashy/waterline#308. Please continue the discussion there, closing this as duplicate Please reopen if you feel there is a specific `sails-mongo` needs a different solution.

**Comment by pavel-anisimov** [comment]
+1. Something like that would be just perfect. It could go even further... Accept array of populate fields... go on multiple levels...
```
findOne().populate(['assignment', 'project']).populate(['assignment.location', 'project.location']);
findOne().populate('project').populate('project.assignment').populate('project.assignment.location').populate('project.assignment.supervisor');
```

**Comment by sailsbot** [comment]
Thanks for posting, @Haks1.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by tarlepp** [comment]
What is the real status of this issue? imho it's shame that @sailsbot closes these issues and users cannot re-open these if the issue isn't yet closed...

**Comment by devinivy** [comment]
@tarlepp agreed.  In this case #308 and PR #1052 can probably replace this issue.  If you disagree, I'd be happy to reopen.

**Comment by BaggersIO** [comment]
+1 - this would be a great feature ...

**Comment by pavel-anisimov** [comment]
I've heard that this will be fixed in next release of Waterline... I hope it is not just a rumor

**Comment by HiNeX** [comment]
+1 - Very useful feature

**Comment by Name-less** [comment]
+1, not any news for this awesome feature ?

**Comment by leomperes** [comment]
+1 Very important feature!

**Comment by tarlepp** [comment]
Any updates for this? @particlebanana

**Comment by joeryy** [comment]
Any updates on this?
